apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -53,5 +74,46 @@ func validateRAID(r *RAIDConfig) error {
 		return fmt.Errorf("hardwareRAIDVolumes and softwareRAIDVolumes can not be set at the same time")
 	}
 
+	if bmcAccess == nil || bmcAccess.RAIDInterface() == "no-raid" {
+		return fmt.Errorf("the 'raid' field for %s is not supported", bmcAccess.Type())
+	}
+
+	return nil
+}
+
+func validateFirmware(firmware *FirmwareConfig, bmcAccess bmc.AccessDetails) error {
+	if firmware == nil {
+		return nil
+	}
+
+	if bmcAccess == nil {
+		return fmt.Errorf("firmware settings are not supported")
+	}
+
+	_, err := bmcAccess.BuildBIOSSettings((*bmc.FirmwareConfig)(firmware))
```

Comment:
It looks like firmware is already in `*bmc.FirmwareConfig` type, maybe we don't have to cast it?

Reply:
We still do this because parameter passed in is `Spec.Firmware`，it is in the v1alpha1 package.

Reply:
if bmcAccess is nil, this will panic.